### ISSUE TITLE
Allow more fine-grained suppression of `ConstructorInvokesOverridable`

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#656]
Consider the following code:
```java
final class Container {
    @SuppressWarnings("ConstructorInvokesOverridable")
    abstract static class Test1 {
        private final Class<?> type = myType();

        abstract Class<?> myType();
    }

    abstract static class Test2 {
        @SuppressWarnings("ConstructorInvokesOverridable")
        private final Class<?> type = myType();

        abstract Class<?> myType();
    }

    abstract static class Test3 {
        @SuppressWarnings("ConstructorInvokesOverridable")
        Test3() {
            myType();
        }

        abstract Class<?> myType();
    }
}
```

Running with `-Xep:ConstructorInvokesOverridable:ERROR` yields:
```
Container.java:11: error: [ConstructorInvokesOverridable] Constructors should not invoke overridable methods.
        private final Class<?> type = myType();
                                            ^
    (see http://errorprone.info/bugpattern/ConstructorInvokesOverridable)
Container.java:19: error: [ConstructorInvokesOverridable] Constructors should not invoke overridable methods.
            myType();
                  ^
    (see http://errorprone.info/bugpattern/ConstructorInvokesOverridable)
2 errors
```

As you can see, no error is emitted for `Test1`. But for the other two classes `@SuppressWarnings("ConstructorInvokesOverridable")` has no effect.

For classes with many fields or much code in initializer blocks/constructors, it'd be nice if `ConstructorInvokesOverridable` could be suppressed at a scope smaller than the whole class. (Refactoring the affected classes would be even better, of course, but alas.)

**Comment by Stephan202** [comment]
A fix for this issue is part of #1449.
